### PR TITLE
Home work 2: Numbers

### PR DESCRIPTION
**Pull request by mylak**

### CODE REVIEW

File: Exercises/data-types-overview/data-types-overview.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// # Data types
+// 1. Declare a variable on each type using two options: plain creation, creation with constructor.
+
+// plain creation:
+let stringVariable = "Hello World!";
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: always use `const` when variable not going to be change

File: Exercises/data-types-overview/data-types-overview.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+let booleanVariable = true;
+let nullVariable = null;
+let undefinedVariable = undefined;
+let objectVariable = {
+  name: "John",
+  age: 30
+}
+
+// creation with constructor:
+let stringConstructorVariable = new String ("The string from constructor");
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: no space between class and `(`

File: Exercises/data-types-overview/data-types-overview.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+let boolinConstructorVariable = new Boolean (false);
+let objectConstructorVariable = new Object ({company:"Ciklum"});
+
+// 2. Use operator typeof on each of them.
+
+// plain creation:
+console.log (typeof (stringVariable));
+console.log (typeof (numberVariable));
+console.log (typeof (booleanVariable));
+console.log (typeof (nullVariable));
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: no need to add `(`

File: Exercises/data-types-overview/data-types-overview.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+};
+
+let objTwo = objOne;
+
+objOne.x = 3;
+objTwo.z = 5;
+
+console.log(objOne); 
+console.log(objTwo);
+console.log(objOne===objTwo);
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: spaces arounf `===`

File: Exercises/data-types-overview/data-types-overview.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+console.log (stringBackticks);
+
+// # Additional
+// 1. Declare an array containing different data types.
+let arrayVariables = ["string", 228, true, null, undefined, {name:"Jack", age: 23}];
+
+console.log(arrayVariables);
+
+// 2. Iterate through it and collect amount of each data type to object.
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: are you skipping 2?

File: Exercises/numbers/numbers.js [hunk 6]
```diff
@@ -1,3 +1,58 @@
 function extractNumbers (text) {
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: some mistake, delete this function you have it below

File: Exercises/numbers/numbers.js [hunk 7]
```diff
@@ -1,3 +1,58 @@
...
+console.log(binaryVar == octalVar);
+console.log(octalVar == hexadecimalVar);
+console.log(binaryVar == hexadecimalVar);
+
+// 2. Implement function `extractNumbers` which will leave only numbers in passed string.
+
+let stringWithNumbers = "123456testQwerty";
+
+function extractNumbers (text) {
+  return parseInt(text);
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: maybe float numbers in string

File: Exercises/numbers/numbers.js [hunk 8]
```diff
@@ -1,3 +1,58 @@
 function extractNumbers (text) {
   return null;
-}
\ No newline at end of file
+}
+
+// # Numbers
+// 1. Assign by hand same value to 3 variables in binary, octal and hexadecimal format.
+
+let binaryVar = "11110011110010000";
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: always use `const` when variable not going to be change

File: Exercises/data-types-overview/data-types-overview.js (hunk 4 above)
Comment (mylak) [reply Exercises/data-types-overview/data-types-overview.js]: fixed

File: Exercises/data-types-overview/data-types-overview.js (hunk 5 above)
Comment (mylak) [reply Exercises/data-types-overview/data-types-overview.js]: yeah, I skipped it and will return to it later

File: Exercises/numbers/numbers.js (hunk 6 above)
Comment (mylak) [reply Exercises/numbers/numbers.js]: fixed

File: Exercises/numbers/numbers.js (hunk 7 above)
Comment (mylak) [reply Exercises/numbers/numbers.js]: fixed

File: Exercises/numbers/numbers.js (hunk 8 above)
Comment (mylak) [reply Exercises/numbers/numbers.js]: but I parse the values of these variables to integer and if I will use 'const' there will be an error

File: Exercises/data-types-overview/data-types-overview.js (hunk 3 above)
Comment (mylak) [reply Exercises/data-types-overview/data-types-overview.js]: fixed

File: Exercises/data-types-overview/data-types-overview.js (hunk 2 above)
Comment (mylak) [reply Exercises/data-types-overview/data-types-overview.js]: fixed

File: Exercises/data-types-overview/data-types-overview.js (hunk 1 above)
Comment (mylak) [reply Exercises/data-types-overview/data-types-overview.js]: fixed. But I thought `const` is an old syntaxis

File: Exercises/numbers/numbers.js [hunk 9]
```diff
@@ -1,3 +1,54 @@
...
-}
\ No newline at end of file
+  return parseFloat(text);
+}
+
+console.log(extractNumbers(stringWithNumbers));
+
+// 3. Declare a variable with decimal value and convert to binary, octal and hexadecimal.
+
+let num = 124816;
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: I think here you can use `const` as well

File: Exercises/numbers/numbers.js (hunk 8 above)
Comment (lizarusi) [reply Exercises/numbers/numbers.js]: my bad, I think in this case name of variable not good, as you called it `binary` and later on it stores decimal value

**Pull request merged**